services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3940,6 +3963,33 @@ public void topicCleanupWhenPushComplete(String clusterName, String storeName, i
           true,
           expectedMinCompactionLagMs,
           expectedMaxCompactionLagMs > 0 ? Optional.of(expectedMaxCompactionLagMs) : Optional.empty());
+
+      // Compaction settings should also be applied to corresponding materialized view topics
+      Map<String, ViewConfig> viewConfigs = store.getVersionOrThrow(versionNumber).getViewConfigs();
+      if (viewConfigs != null && !viewConfigs.isEmpty()) {
+        Set<String> viewTopicsToUpdate = new HashSet<>();
+        for (ViewConfig rawViewConfig: viewConfigs.values()) {
+          if (MaterializedView.class.getCanonicalName().equals(rawViewConfig.getViewClassName())) {
+            viewTopicsToUpdate.addAll(
+                ViewUtils
+                    .getVeniceView(
+                        rawViewConfig.getViewClassName(),
+                        new Properties(),
+                        storeName,
+                        rawViewConfig.getViewParameters())
+                    .getTopicNamesAndConfigsForVersion(versionNumber)
+                    .keySet());
+          }
+        }
+        for (String topic: viewTopicsToUpdate) {
+          PubSubTopic viewTopic = pubSubTopicRepository.getTopic(topic);
+          getTopicManager().updateTopicCompactionPolicy(
+              viewTopic,
+              true,
+              expectedMinCompactionLagMs,
+              expectedMaxCompactionLagMs > 0 ? Optional.of(expectedMaxCompactionLagMs) : Optional.empty());
+        }
```

Comment:
I see some duplicates around topic compaction policy updates, so can we get rid of duplicate codes?